ci: run windows CI jobs on PRs

josephperrott opened the pull request:
Previously windows CI jobs were only run on upstream branches, with the addition
of larger Windows executors as well as the improvement of setup speed in the
windows environment setup script allows for the windows tests to pass in a
reasonable timeframe.

josephperrott committed:
ci: run windows CI jobs on PRs

Previously windows CI jobs were only run on upstream branches, with the addition
of larger Windows executors as well as the improvement of setup speed in the
windows environment setup script allows for the windows tests to pass in a
reasonable timeframe.

sonukapoor commented on .circleci/windows-env.ps1:
```diff
@@ -2,8 +2,8 @@
 # Install MSYS2 and packages
-choco install msys2 --version 20180531.0.0 --no-progress --package-parameters "/NoUpdate"
-C:\tools\msys64\usr\bin\bash.exe -l -c "pacman --needed --noconfirm -S zip unzip patch diffutils git"
+choco install msys2 --version 20200903.0.0 --no-progress --package-parameters "/NoUpdate"
```
Does `choco` automatically install `msys2` into the `C:\tools\msys64 folder`? I have it installed in my `C` folder only.

josephperrott replied:
It installs to the value of `Get-ToolsLocation`, which at least on CircleCI is `C:\tools`

sonukapoor replied:
Got it 👍

josephperrott committed:
ci: setup windows from scratch

Rather than setting up windows by relying on attaching the saved workspace
failes from the previous step, instead checkout and install the yarn items
within the windows steps.  Additionally, since the bazel remote cache is
used and relied on, saving the cached results of the bazel runs to be resumed
on subsequent runs does not provide enough value to make it worth the time
consumed.

gkalpak commented on .circleci/config.yml:
```diff
@@ -184,22 +182,12 @@ commands:
     description: Setup windows node environment
     steps:
       # Use the Linux workspace directly, as it already has checkout, rebased and node modules.
```
This comment is no longer valid.

gkalpak reviewed (approved):
🚀

IgorMinar reviewed (approved):
Overall this looks good except that in the last CI run the windows VE job took 20min (most of all the jobs) so it seems that this job will become a bottleneck for the CI.

Interestingly, the 20min long job run already benefited from many cache hits, which means that on average the job duration might be even worse.

Let's get this PR in, but let's also monitor the job duration to understand the overall impact of this change and consider further changes to the windows setup if necessary.

josephperrott committed:
fixup! ci: setup windows from scratch
